### DISCUSSION TITLE
Best practices for securing Appwrite

### DISCUSSION THREAD

**Discussion by issasaleh** [appwrite/appwrite#5862] [Q&A]
Hi,

I'm new to this, I'm building an Android app and want to completely protect the appwrite endpoint from anyone, even listing documents, is there a way to prevent anyone from using the endpoint other than just my app?

thanks,

**Comment by rafagazani** [comment]
it sure is possible. You will need to set permissions per collection or document. See official doc https://appwrite.io/docs/permissions
If possible join discord https://appwrite.io/discord
(it is the most receptive community I know, there is no question without an answer)

**Reply by issasaleh** [reply]
This thing will not work for me, because I must put the read permissions to all guests, here is the problem. If someone managed to get my API, he can pull all the data from me and make an application similar to the one I have, with free data that he stole for me. Is there anything like the one in Firebase, (SHA-1 Signature verification)

**Reply by stnguyen90** [reply]
> Is there anything like the one in Firebase, (SHA-1 Signature verification)

@issasaleh, for that, make sure to 👍🏼 this issue: https://github.com/appwrite/appwrite/issues/4791
